NationalJournal/vue-treeselect
# Customizations

### Changes

Added NJ customizations onto riophae/vue-treeselect repo.

- Added `aria-label` prop to input.
- Added `exclusive` prop to force one branch expanded at a time.
- Added `showSelectedChildrenCount` prop to display count of selected items under a branch node
- Automatically collapse branch node if it's selected.

### Definitions

_Branch node means top level item with children_

### Testing:

- Clone repo, make sure you're on `updates` branch.
- `yarn install`
- `PORT=8081 yarn dev`
- Go to http://localhost:8081/#more-features
- Testing __showSelectedChildrenCount__:
   - Select `Show children count` and `Show selected children count` options, make sure `Multi-select` `Clearable` `Searchable` `Open on click` are selected as well.
   - When you open dropdown, you should see number of child elements `(3)` next to branch nodes.
   - When you expand one of branch nodes, you should still see `(3)` next to its label.
   - Select one of the children, let's say `A -> AA`
      - `A` will get partial selected [-], and `AA` should display check.
      - `A` label should still display `(3)`.
   - Collapse `A`. Label should now display `(1/3 selected)`
   - Expand `B`.
      -  Click `B` to select it and all child nodes.
      - `B` label should display `(3/3 selected)` and it should collapse automatically.
   - Click `C`. It should now also display `(3/3 selected)` next to it, and it should stay collapsed.
- Testing __showSelectedChildrenCount while searching__:
   - _Continuing off previous step with some items already selected_, type `A` into dropdown list to search for nodes containing `A`
      - You should NOT see how many items are selected, instead it should just display how many children are found under each branch node. 
- Testing __exclusive prop__:
   - Reload the page to reset demo dropdown, navigate to `More Features` again.
   - Select `Exclusive` option.
   - Expand `A` by clicking arrow next to the label. It should display its children.
   - Expand `B`. `A` should collapse, and `B` should be expanded now.
   - Expand `A`. `B` should collapse, and `A` should be expanded again.
- Testing __automatically collapse branch node if it's selected__:
   - Reload the page to reset demo dropdown, navigate to `More Features` again.
   - Expand `A` by clicking arrow next to the label. It should display its children.
   - Select `A` by either clicking checkbox or clicking on label.
      - `A` should collapse automatically.
   - Expand `B` by clicking arrow next to the label. It should display its children.
      - Select all 3 children of `B`.
      - `B` should collapse automatically once all children have been selected.
